modules/tensor_mechanics/src/materials/ComputeCrackPropagationHeatEnergy.C

```diff
@@ -0,0 +1,71 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ComputeCrackPropagationHeatEnergy.h"
+
+template <>
+InputParameters
+validParams<ComputeCrackPropagationHeatEnergy>()
+{
+  InputParameters params = validParams<Material>();
+  params.addParam<std::string>("base_name",
+                               "Optional parameter that allows the user to define "
+                               "multiple mechanics material systems on the same "
+                               "block, i.e. for multiple phases");
+  params.addClassDescription("Crack propagation heat energy density = - (dPsi/dc) * (dc/dt) "
+                             "is the energy dissipated due to damage increase "
+                             "Psi is the free energy of the phase-field fracture model "
+                             "defined as Psi = (1 - c)^2 * G0_pos + G0_neg "
+                             "c is the order parameter for damage, continuous between 0 and 1 "
+                             "0 represents no damage, 1 represents fully cracked "
+                             "G0_pos and G0_neg are the positive and negative components "
+                             "of the specific strain energies");
+  params.addRequiredCoupledVar("c", "Phase field damage variable");
+  params.addCoupledVar(
+      "displacements",
+      "The displacements appropriate for the simulation geometry and coordinate system");
+  return params;
+}
+
+ComputeCrackPropagationHeatEnergy::ComputeCrackPropagationHeatEnergy(
+    const InputParameters & parameters)
+  : DerivativeMaterialInterface<Material>(parameters),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _c(coupledValue("c")),
+    _c_old(coupledValueOld("c")),
+    _G0_pos(getMaterialPropertyByName<Real>(_base_name + "G0_pos")),
+    _dG0_pos_dstrain(getMaterialPropertyByName<RankTwoTensor>(_base_name + "dG0_pos_dstrain")),
+    _crack_propagation_heat(declareProperty<Real>(_base_name + "crack_propagation_heat")),
+    _dcrack_propagation_heat_dstrain(
```

Comment:
You're already hooked up with the `DerivativeMaterialInterface`, so you should use the nifty `declarePropertyDerivative` function here when declaring these material properties that are derivatives with respect to solution variables.

Reply:
I have used declarePropertyDerivative for _dcrack_propagation_heat_dc because it is a derivative with respect to a variable. _dcrack_propagation_heat_dstrain is a derivative with respect to another property (strain). I am not sure if in this last case I can use declarePropertyDerivative.